packages/dcos-integration-test/extra/test_endpoints.py

```diff
@@ -76,10 +77,15 @@ def test_if_uiconfig_is_available(cluster):
 
 
 def test_if_dcos_history_service_is_up(cluster):
-    r = cluster.get('/dcos-history-service/ping')
 
-    assert r.status_code == 200
-    assert 'pong' == r.text
+    @retry(stop_max_delay=20000, wait_fixed=500)
```

Comment:
So this is covering a different flakiness the switch to gunicorn found / exposed. So dcos-history-service up checks that one instance somewhere in the cluster is up. This test then sometimes hits other instances in the cluster, which sometimes are up but have no data.